submissions/ses3332002/friends-app/app.js

```diff
@@ -0,0 +1,76 @@
+"use strict";
+const hamburgerMenu = document.querySelector(".hamburger_menu");
+const mainArea = document.querySelector("main");
+const optionForm = document.forms.sorting_n_filtering;
+let receivedData = [];
+let dataToRender = [];
+let filterCriterionSex = () => true;
+let filterCriterionAge = () => true;
+let filterCriterionName = () => true;
+let sortCriterion = () => true;
+as_is.checked = true;
+all.checked = true;
+age_from.value = "";
+age_to.value = "";
+search_name.value = "";
+hamburgerMenu.addEventListener("click", () => hamburgerMenu.classList.toggle("hamburger_menu-opened"));
+optionForm.addEventListener("input", changeData);
+
+fetch("https://randomuser.me/api/?nat=gb,us,fr&results=60&inc=gender,name,location,email,picture,dob")
+  .then((response) => {
+    return response.json();
+  })
+  .then((data) => {
+    receivedData = data.results;
+    dataToRender = receivedData.slice().filter(filterCriterionSex).filter(filterCriterionAge).filter(filterCriterionName).sort(sortCriterion);
```

Comment:
Do you really need to do all this filtering and sorting on the first render?

Reply:
Yes. If the first render is going without any filters or sorting then after user choose any "neutral" filter (like "as is" or "all") cards array is changing.

Reply:
Yes

Reply:
so why you do all this filtering and sorting on the first render?

Reply:
As I said before, if the first render is going without any filters or sorting then after user choose any "neutral" filter (like "as is" or "all") cards array is changing. I couldn't find out why, so I solved this problem "at the root".
Now I removed from Demo this filtering on the first render, so you can see it yourself: reload page, remember some faces, choose "man" and then - choose "all". They will be another "all", then after reload :)

Reply:
I've found a mistake and remade it. Now there's no filtering and sorting on the first render.